### DISCUSSION TITLE
Add a guard that is executed before redirect

### DISCUSSION THREAD

**Discussion by guizoxxv** [vuejs/router#1713] [Ideas]
I recently strugled to execute async code before a redirect as you can see in [this discussion](https://stackoverflow.com/questions/75569000/how-to-execute-async-code-before-redirect-on-vue-router/75569159?noredirect=1#comment133327453_75569159). I was able to achieve this redirecting inside the `beforeEnter` instead of in the `redirect` attribute.

Althought it works, this approach still requires a component atrribute even thought I'm always redirecting. It would be interesting to have a guard that is executed before the redirect (beforeRedirect or a more generic name). I believe it would make the code more intuitive instead of setting a `component` just to fill the attribute.

Thanks.

**Comment by posva** [comment]
If you need to execute async code before redirecting, that's a navigation guard. There is already plenty of ways to add them.

If you are sure your `beforeEnter()` will always redirect, then it's fine to set a placeholder component or to add a `@ts-expect-error` on top of `component: null`
